### DISCUSSION TITLE
What's the best way to handle reply-to messages.

### DISCUSSION THREAD

**Discussion by erwanmace** [MessageKit/MessageKit#1691] [Q&A]
I would like to add a view inside the bubble (on top) with the message that this current message is replying to. 

I find the process if subclassing the Message cell extremely complex especially if I need to do it for each type of message: text, image, link preview, etc... 

Same issue if I wanted to include the timestamp inside the bubble. Would I need to subclass all different type of message cells?

**Comment by Kaspik** [comment]
I would simply do completely custom cells. There is no reason why you should subclass any existing cells, these are specific for specific simple things. If you want custom one, simply do custom cell type. :)

**Reply by erwanmace** [reply]
But if my custom cell don’t even subclass the existing one, that means I need to recreate the whole set of text cells, photo cells, link preview cells, etc…. From scratch. 

All I want is to reuse the preset cells (they’re good) but just add a Subview on top in case it’s a reply message. And maybe a subview at the bottom (inside the bubble) for the time stamp. 

Custom cells would still inherit the avatar, top and bottom labels, bubble style, etc… ?

**Reply by erwanmace** [reply]
@Kaspik does that mean that I would need to subclass all type of message views: media message, audio message, text message, link message, etc...  Since they all subclass MessageContentCell, would it make sense to extend (not subclass) MessageContentCell to insert the reply-to-message on top?

Any other easier approach?

**Comment by ElrB** [comment]
I simply do it almost like how iMessage does it, insert two messages (styled differently), the first shows the original message the user is replying to, the second shows the reply message. The first message that reference the original message being replied to is especially styled differently to show that it is not an original message.

**Comment by erwanmace** [comment]
@ElrB do you mind sharing a screenshot. 

Are you using the native message view cells or custom ?

i find WhatsApp’s approach more user friendly than iMessage. I just need to insert An additional view in the content view of a message but I’m not sure how to do that and update message size calculator without rewriting the whole message view cell class

**Comment by JuinJuice** [comment]
@erwanmace Hello! Did you find a solution for reply things?
Or you followed the advice to create a custom cell for all types?
